eurostag-step-up-transformer/src/main/java/eu/itesla_project/iidm/ddb/eurostag/EurostagDDB.java

```diff
@@ -26,12 +28,21 @@
 
     EurostagDDB(List<Path> ddbDirs) throws IOException {
         for (Path ddbDir : ddbDirs) {
+            if (Files.isSymbolicLink(ddbDir)) {
+                ddbDir = Files.readSymbolicLink(ddbDir);
+            }
             if (!Files.exists(ddbDir) && !Files.isDirectory(ddbDir)) {
                 throw new IllegalArgumentException(ddbDir + " must exist and be a dir");
             }
-            Files.walkFileTree(ddbDir, new SimpleFileVisitor<Path>() {
+            Files.walkFileTree(ddbDir, EnumSet.of(FileVisitOption.FOLLOW_LINKS), Integer.MAX_VALUE, new SimpleFileVisitor<Path>() {
                 @Override
                 public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    if (Files.isSymbolicLink(file)) {
+                        file = Files.readSymbolicLink(file);
+                        if (Files.isDirectory(file)) {
+                            Files.walkFileTree(file, EnumSet.of(FileVisitOption.FOLLOW_LINKS), Integer.MAX_VALUE, this);
+                        }
+                    }
```

Comment:
Refactor this code: you probably should add « else if » somewhere to avoid useless test

Reply:
Up!